### PR TITLE
Added virtualenv

### PR DESCRIPTION
**Pull request by appi147**
#238

### CODE REVIEW

**Comment by appi147** [comment]
@StefanosChaliasos Please review

**Comment by appi147** [comment]
@krishnapaul242 Please verify if this fixes #224

File: setup.sh [hunk 1]
```diff
@@ -38,8 +38,12 @@ if [[ "$OS" == "Fedora" ]]; then
 elif [[ "$OS" == "Ubuntu" ]] || [[ "$OS" == "LinuxMint" ]]; then
   sudo apt-get install ffmpeg python-imdbpy python-notify2
   sudo apt-get install python-dbus libssl-dev libffi-dev
-  sudo -H pip install -r requirements.txt
   sudo apt-get install chromium-chromedriver
+  sudo pip install virtualenv
+  virtualenv env
+  virtualenv -p /usr/bin/python2.7 env
+  source env/bin/activate
+  pip install -r requirements.txt
```
Comment (StefanosChaliasos) [inline setup.sh]: you forgot sudo here

File: setup.sh [hunk 2]
```diff
@@ -38,8 +38,12 @@ if [[ "$OS" == "Fedora" ]]; then
 elif [[ "$OS" == "Ubuntu" ]] || [[ "$OS" == "LinuxMint" ]]; then
   sudo apt-get install ffmpeg python-imdbpy python-notify2
   sudo apt-get install python-dbus libssl-dev libffi-dev
-  sudo -H pip install -r requirements.txt
   sudo apt-get install chromium-chromedriver
+  sudo pip install virtualenv
+  virtualenv env
+  virtualenv -p /usr/bin/python2.7 env
+  source env/bin/activate
```
Comment (StefanosChaliasos) [inline setup.sh]: it didn't work for me

File: setup.sh (hunk 1 above)
Comment (appi147) [reply setup.sh]: `sudo` in which line?

Comment (StefanosChaliasos) [reply setup.sh]: 46

Comment (appi147) [reply setup.sh]: `sudo` will install packages systemwide but we need to install packages in virtual environment

File: setup.sh (hunk 2 above)
Comment (appi147) [reply setup.sh]: can you give me errors(screenshot or log)

File: setup.sh (hunk 1 above)
Comment (StefanosChaliasos) [reply setup.sh]: Yeah you are right. The problem is that some packages need sudo to install. You can try something like [that](https://askubuntu.com/questions/234758/how-to-use-a-python-virtualenv-with-sudo)

File: setup.sh (hunk 2 above)
Comment (StefanosChaliasos) [reply setup.sh]: error_message

Comment (appi147) [reply setup.sh]: Is it working now?

**Comment by appi147** [comment]
@StefanosChaliasos Is it working now?

File: setup.sh (hunk 1 above)
Comment (appi147) [reply setup.sh]: As pointed out in the article, if we need an external script to run as superuser, we have to follow that method, but pip is already created once we create a virtual environment.

**Comment by StefanosChaliasos** [comment]
It worked for me (Ubuntu)

**Comment by krishnapaul242** [comment]
No, It didn't work.

**Comment by appi147** [comment]
I'm on Ubuntu 17. 04 and tested on unity as well as xfce desktop environments

**Comment by appi147** [comment]
@krishnapaul242 if you are on another desktop environment please let us know

**Pull request merged**